### ISSUE TITLE
recusing into \text

### ISSUE THREAD

**Issue by kohlhase** [KWARC/sTeX#37]
In ```smglom/numberfields/source/intarith.de.tex```, we have 
```LaTeX
    \[\inttimes[cdot]{a,b}=\piecewise{
      \piece[de]{\nattimes[cdot]{\absolutevalue{a},\absolutevalue{b}}}
                      {\text{$\minset{a,b}\NaturalNumbers$ or $\minset{a,b}\NegativeIntegerNumbers$}}
      \otherwise[de]{\uminus{\nattimes[cdot]{\absolutevalue{a},\absolutevalue{b}}}}}
    \]
```
the important part of this is 
```LaTeX
\text{$\minset{a,b}\NaturalNumbers$ or $\minset{a,b}\NegativeIntegerNumbers$}
```
which generates 
```XML
	    <om:OMSTR>
	      <om:OMSTR xml:id="S0.Ex2.m1.m1.om">a,bminset&#x2061;a,b, NaturalNumbers</om:OMSTR>
	      <om:OMSTR> or </om:OMSTR>
	      <om:OMSTR xml:id="S0.Ex2.m1.m2.om">a,bminset&#x2061;a,b, NegativeIntegerNumbers</om:OMSTR>
	    </om:OMSTR>
```
which is not what I want. For the moment I would settle for 
```XML
<om:OMFOREIGN encoding="application/omdoc+xml">
  <om:OMA>
     <om:OMS cd="set" name="minset"/>
     <om:OMV name="a" xml:id="XM5.om"/>
     <om:OMV name="b" xml:id="XM6.om"/>
     <om:OMS cd="naturalnumbers" name="natural-number"/>
   </om:OMA>
</om:OMA>
or 
<om:OMSTR>
  <om:OMA>
     <om:OMS cd="set" name="minset"/>
     <om:OMV name="a" xml:id="XM5.om"/>
     <om:OMV name="b" xml:id="XM6.om"/>
     <om:OMS cd="naturalnumbers" name="negative-natural-number"/>
   </om:OMA>
</om:OMA>
</om:OMSTR>
``` 
I am not sure how to best specify the encoding here, but OMDoc should do fine. 

I have not quite been able to test extensively, since my LaTeXML installation is quite broken, but I think this is a post-processing problem. I ran the following little experiment: 
```LaTeX
\documentclass{article}
\usepackage{amstext}
\begin{document}
$f(\text{$\alpha\in N$ or $\alpha\in M$})$
\end{document}
``` 
to have some math embedded in text embedded in math. This runs through ```pdflatex``` beautifully (somewhat to my surprise). And ```latexml``` gives me 
```XML
<?xml version="1.0" encoding="UTF-8"?>
<?latexml searchpaths="/Users/kohlhase/tmp"?>
<?latexml class="article"?>
<?latexml package="amstext"?>
<?latexml RelaxNGSchema="LaTeXML"?>
<document xmlns="http://dlmf.nist.gov/LaTeXML">
  <resource src="LaTeXML.css" type="text/css"/>
  <resource src="ltx-article.css" type="text/css"/>
  <para xml:id="p1">
    <p>
    <Math mode="inline" xml:id="p1.m1" tex="f(\text{$\alpha\in N$ or $\alpha\in M$})"
	  text="f * [∈αN or ∈αM]">
      <XMath>
	<XMApp>
	  <XMTok meaning="times" role="MULOP">⁢</XMTok>
	  <XMTok possibleFunction="yes" role="UNKNOWN" font="italic">f</XMTok>
	  <XMDual>
	    <XMRef idref="XM1"/>
	    <XMWrap>
	      <XMTok role="OPEN" stretchy="false">(</XMTok>
	      <XMText xml:id="XM1">
		<Math mode="inline" tex="\alpha\in N" text="alpha element-of N" xml:id="p1.m1.m1">
		  <XMath>
		    <XMApp>
		      <XMTok meaning="element-of" name="in" role="RELOP">∈</XMTok>
		      <XMTok name="alpha" role="UNKNOWN" font="italic">α</XMTok>
		      <XMTok role="UNKNOWN" font="italic">N</XMTok>
		    </XMApp>
		  </XMath>
		</Math>
		or
		<Math mode="inline" tex="\alpha\in M" text="alpha element-of M" xml:id="p1.m1.m2">
		  <XMath>
		    <XMApp>
		      <XMTok meaning="element-of" name="in" role="RELOP">∈</XMTok>
		      <XMTok name="alpha" role="UNKNOWN" font="italic">α</XMTok>
		      <XMTok role="UNKNOWN" font="italic">M</XMTok>
		    </XMApp>
		  </XMath>
		</Math>
	      </XMText>
	      <XMTok role="CLOSE" stretchy="false">)</XMTok>
	    </XMWrap>
	  </XMDual>
	</XMApp>
      </XMath>
    </Math>
    </p>
  </para>
</document>
```
which is all we can hope for. But then the postprocessors give me trouble: for HTML I get 
```XML
<div id="p1" class="ltx_para">
  <p class="ltx_p">
  <math id="p1.m1" class="ltx_Math" alttext="f(\text{$\alpha\in N$ or $\alpha\in M$})"
	display="inline">
    <mrow>
      <mi>f</mi>
      <mo>â¢</mo>
      <mrow>
	<mo stretchy="false">(</mo>
	<mrow id="XM1">
	  <mrow id="XM1a">
	    <mi id="XM1b">Î±</mi>
	    <mo id="XM1c">â</mo>
	    <mi id="XM1d">N</mi>
	  </mrow>
	  <mtext id="XM1e">Â orÂ </mtext>
	  <mrow id="XM1f">
	    <mi id="XM1g">Î±</mi>
	    <mo id="XM1h">â</mo>
	    <mi id="XM1i">M</mi>
	  </mrow>
	</mrow>
	<mo stretchy="false">)</mo>
      </mrow>
    </mrow>
  </math>
  </p>
</div>
```
where the encoding is off (but that may be my installation). And in sTeX, I get the non-standard behavior above. 

I would appreciate any help.

**Comment by dginev** [comment]
Should be fixed by 8c8076ccab071b3875f9dfb4dec414bf8865f13a, reopen if otherwise.

**Comment by dginev** [comment]
And indeed, pointing me to ```pmml_text_aux``` gave some clarity, especially after I _really_ understood the trickery it was doing.
